pkg/image/dockerv1client/client.go

```diff
@@ -34,46 +33,53 @@ var (
 	ImageScheme = runtime.NewScheme()
 )
 
+// ConvertImageToDockerImage converts an object of type *godockerclient.Image to *docker10.DockerImage
+func ConvertImageToDockerImage(in *godockerclient.Image, out *docker10.DockerImage, s conversion.Scope) error {
+	if err := s.Convert(&in.Config, &out.Config, conversion.AllowDifferentFieldTypeNames); err != nil {
+		return err
+	}
+	if err := s.Convert(&in.ContainerConfig, &out.ContainerConfig, conversion.AllowDifferentFieldTypeNames); err != nil {
+		return err
+	}
+	out.ID = in.ID
+	out.Parent = in.Parent
+	out.Comment = in.Comment
+	out.Created = metav1.NewTime(in.Created)
+	out.Container = in.Container
+	out.DockerVersion = in.DockerVersion
+	out.Author = in.Author
+	out.Architecture = in.Architecture
+	out.Size = in.Size
+	return nil
+}
+
+// ConvertDockerImageToImage converts an object of type *docker10.DockerImage to *godockerclient.Image
+func ConvertDockerImageToImage(in *docker10.DockerImage, out *godockerclient.Image, s conversion.Scope) error {
+	if err := s.Convert(&in.Config, &out.Config, conversion.AllowDifferentFieldTypeNames); err != nil {
+		return err
+	}
+	if err := s.Convert(&in.ContainerConfig, &out.ContainerConfig, conversion.AllowDifferentFieldTypeNames); err != nil {
+		return err
+	}
+	out.ID = in.ID
+	out.Parent = in.Parent
+	out.Comment = in.Comment
+	out.Created = in.Created.Time
+	out.Container = in.Container
+	out.DockerVersion = in.DockerVersion
+	out.Author = in.Author
+	out.Architecture = in.Architecture
+	out.Size = in.Size
+	return nil
+}
```

Comment:
This is bad, the reason this was internal was to ensure nobody will use that, your approach suddenly makes those conversion functions public, at least make them private.

Reply:
updated